File: android/app/src/main/java/fr/magiclockscreen/android/MagicPrefs.java
package fr.magiclockscreen.android;

import android.content.Context;
import android.content.SharedPreferences;

public final class MagicPrefs {
    private static final String NAME = "magic_lockscreen_prefs";

    private MagicPrefs() {}

    private static SharedPreferences prefs(Context context) {
        return context.getSharedPreferences(NAME, Context.MODE_PRIVATE);
    }

    public static String backend(Context context) {
        return prefs(context).getString("backend", "");
    }

    public static String token(Context context) {
        return prefs(context).getString("token", "");
    }

    public static int intervalSeconds(Context context) {
        return prefs(context).getInt("interval", 2);
    }

    public static int durationMinutes(Context context) {
        return prefs(context).getInt("duration", 10);
    }

    public static String lastHash(Context context) {
        return prefs(context).getString("lastHash", "");
    }

    public static void setLastHash(Context context, String hash) {
        prefs(context).edit().putString("lastHash", hash).apply();
    }

    public static void saveConfig(Context context, String backend, String token, int interval, int duration) {
        String cleanBackend = backend == null ? "" : backend.trim();
        while (cleanBackend.endsWith("/")) {
            cleanBackend = cleanBackend.substring(0, cleanBackend.length() - 1);
        }

        prefs(context).edit()
                .putString("backend", cleanBackend)
                .putString("token", token == null ? "" : token.trim())
                .putInt("interval", Math.max(1, Math.min(interval, 60)))
                .putInt("duration", Math.max(1, Math.min(duration, 240)))
                .apply();
    }
}

File: android/app/src/main/java/fr/magiclockscreen/android/MagicPrefs.kt
package fr.magiclockscreen.android

import android.content.Context

object MagicPrefs {
    private const val NAME = "magic_lockscreen_prefs"

    private fun prefs(context: Context) =
        context.getSharedPreferences(NAME, Context.MODE_PRIVATE)

    fun backend(context: Context): String =
        prefs(context).getString("backend", "") ?: ""

    fun token(context: Context): String =
        prefs(context).getString("token", "") ?: ""

    fun intervalSeconds(context: Context): Int =
        prefs(context).getInt("interval", 2)

    fun durationMinutes(context: Context): Int =
        prefs(context).getInt("duration", 10)

    fun lastHash(context: Context): String =
        prefs(context).getString("lastHash", "") ?: ""

    fun saveConfig(context: Context, backend: String, token: String, interval: Int, duration: Int) {
        prefs(context).edit()
            .putString("backend", backend.trim().trimEnd('/'))
            .putString("token", token.trim())
            .putInt("interval", interval.coerceIn(1, 60))
            .putInt("duration", duration.coerceIn(1, 240))
            .apply()
    }

    fun setLastHash(context: Context, hash: String) {
        prefs(context).edit().putString("lastHash", hash).apply()
    }
}
